src/create-theme-provider.js

```diff
@@ -0,0 +1,92 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const isFunction = require('is-function');
```

Comment:
Unsure, I'd rather be safe than sorry? Is there a canonical reference for this? I feel like the `is-function` people put a lot more thought into this.